hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -2846,7 +2846,13 @@ protected PrepareFlushResult internalPrepareFlushCache(WAL wal, long myseqid,
     String s = "Finished memstore snapshotting " + this + ", syncing WAL and waiting on mvcc, " +
         "flushsize=" + totalSizeOfFlushableStores;
     status.setStatus(s);
-    doSyncOfUnflushedWALChanges(wal, getRegionInfo());
+
+    try {
+      doSyncOfUnflushedWALChanges(wal, getRegionInfo());
+    } catch (Throwable t) {
+      status.abort("Sync unflushed WAL changes failed: " + StringUtils.stringifyException(t));
+      fatalForFlushCache(t);
```

Comment:
Mind explaining a bit why here we need to abort the regionserver? We do not need to abort the regionserver when hitting IOException above?

Reply:
The IOException thrown above is not caused by syncing WAL, only the sync failure should abort the RS as early as possible.  And if we only allow aborting RS in the commit step of flushing cache, there will always exit the circumstance that OOM kill, because all failures in the previous preparing step of flushing cache only aborted the flush process, all flushing can not go to the commit step.

Reply:
@Apache9 We currently only abort RS when there are critical problems, such as the WAL stuck? The above IOExceptions can only abort the flush in mem and should not prevent the next flushes, right? Here the abort is caused by the wal stucks who are ahead of the committing flush wal sync stucks. What do you think?